crates/ra_project_model/src/lib.rs

```diff
@@ -286,3 +304,32 @@ fn find_cargo_toml(path: &Path) -> Result<PathBuf> {
     }
     Err(format!("can't find Cargo.toml at {}", path.display()))?
 }
+
+pub fn get_rustc_cfg_options() -> CfgOptions {
+    let mut cfg_options = CfgOptions::default();
+
+    match (|| -> Result<_> {
```

Comment:
I would slightly prefer to introduce an `fn try_get_rustc_cfg_options() -> Result<CfgOptions, std::io::Error>` helper rather than use an immediately invoked closure, but I don't have a strong opinion here

Reply:
There should be `try` block but it's unstable

Reply:
I don't think `try_get_rustc_cfg_options` should be exposed. I think we should not add fake options when failure. Instead, custom cfg options should be applied outside, though it is not implemented yet.